src/assets/php8.4/php8.4.md

```diff
@@ -0,0 +1,226 @@
+---
+title: PHP 8.4 - Discover the Latest and Greatest
+authorName: Isha Baral
+authorAvatar: https://avatars.githubusercontent.com/u/40142570?s=400&u=58ce006ccc626577e503b75788b2e8634a9b7d9c&v=4
+authorLink: https://github.com/ishabaral/
+createdAt: Aug 22, 2024
+tags: php, php8.4, latest
+banner: https://blog.jankaritech.com/src/assets/php8.4/images/php8.4.png
```

Comment:
that image is huge (2.24MB) could we reduce it to a smaller size?

Reply:
you can use https://squoosh.app/

Reply:
or imagemagic on the CLI:
```
sudo apt-get install imagemagick # if its not installed yet
convert php8.4.png php8.4.webp
```

result for me:
```
-rw-rw-r-- 1 artur artur 2,3M Sep  5 08:43 php8.4.png
-rw-rw-r-- 1 artur artur  60K Sep  5 08:43 php8.4.webp

Reply:
should the banner image be of .webp format? or only reducing image size is okay?
The .webp format gives the stylelint error :
```
src/assets/php8.4/images/php8.4.webP
 2:116  ✖  Unknown word  CssSyntaxError
 ```

Reply:
>should the banner image be of .webp format?

not a requirement.

>The .webp format gives the stylelint error :

maybe due to `php8.4.webP` instead of `php8.4.webp`? (might not be though)

Reply:
As far as I know our internal blog doesn't support all image formats. Most frequently .png, .jpg or .gif is used for banner images. I think .png doesn't compress the images as much as .jpg. Try to find a compromise of reducing the file size but still keeping a fairly good quality for people who will read the blog on desktop. Some of the previous blog posts have banner images that were too small to start with or so much compressed that it looks pixelated on big screens...

Reply:
> > should the banner image be of .webp format?
> 
> not a requirement.
> 
> > The .webp format gives the stylelint error :
> 
> maybe due to `php8.4.webP` instead of `php8.4.webp`? (might not be though)

tried webP too, but same error
As @koebel said, it might not support

Reply:
It's also a good practice to give reference to the source of your image unless you created it yourself. I think the only way this is possible at the moment is mentioning the source in a footnote. It seems like we can't modify the alt tag of the banner image with the current version of our blog software

Reply:
that should not be the case. the bog should support all image formats that browser support (because the blog doesn't do any processing to the images - they just get rendered in the browser).

>The .webp format gives the stylelint error :

the problem here is the stylelint (css linter), it should not check the .webp files. we need to ignore .webp file format in the config
https://github.com/JankariTech/blog/blob/7586adb2d080e909479ed183eda2028836cc1536/stylelint.conf.js#L16-L24